### PR TITLE
"Getting Started" docs for TS SDK

### PR DESCRIPTION
**Pull request by calintje**
**This PR adds documentation for the new TS SDK**

**Details**
- Documentation structure
- Welcome page, including short primer on Solana
- "Whirlpools SDK" category
    - Environment setup, including wallet initialization
    - Conceptual and usage-focused guides on SDK

Screenshot 2024-10-21 115217

### CODE REVIEW

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:52]: We should probably use new web3.js here

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:91]: I would try to keep this as simple as possible. Just do rpc.requestAirdrop and nothing else

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:60]: I would try to keep this as simple as possible. Do we really need to store the key in the env? I would just do `generateKeypair`, check the balance and do an airdrop

File: docs/whirlpool/docs/03-Getting Started/02-Create Pool/01-Splashpool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/02-Create Pool/01-Splashpool.md:34]: I would not refer back to previous functions you made. Assume that whoever is reading this particular page has not read any of the previous pages.

I would just do two things here instead:
```
const wallet = generateKeypair() // from new web3.js
rpc.requestAirdrop(...) // just always request some airdrop
```
Any ready can always swap this logic out for whatever they need (load from file, etc.)

The quicker you can get to the actual code that matters in this case the easier it becomes to read

File: docs/whirlpool/docs/03-Getting Started/02-Create Pool/01-Splashpool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/02-Create Pool/01-Splashpool.md:57]: I like this!

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:26]: The high level sdk is `@orca-so/whirlpools`

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:128]: I wouldn't make actual functions and just write the snippets in script style

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (wjthieme) [reply docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:26]: Do we need ts-node and fs here? The easier we can keep this the easier it becomes for a reader to follow along

File: docs/whirlpool/docs/03-Getting Started/05-Monitor/01-Splash Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/05-Monitor/01-Splash Pool.md:5]: I would expect this to come earlier in the guide, maybe after creating a pool?

File: docs/whirlpool/docs/03-Getting Started/03-Provide Liquidity/01-Open Position/02-CLMM Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Getting Started/03-Provide Liquidity/01-Open Position/02-CLMM Pool.md:2]: Instead of CLMM I would call it `Concentrated Liquidity Pool`

File: docs/whirlpool/docs/05-Legacy SDK/01-Developer Portal.md
Comment (wjthieme) [inline docs/whirlpool/docs/05-Legacy SDK/01-Developer Portal.md:1]: Instead of a link can we add the actual page (so that we can decommission the old developer portal)?

File: docs/whirlpool/package.json
Comment (wjthieme) [inline docs/whirlpool/package.json:14]: Does docusaurus support tw? If so we might want to use that instead of `@emotion`

File: docs/whirlpool/package.json
Comment (wjthieme) [inline docs/whirlpool/package.json:8]: What does this command do (and do we really need it as a script in the package.json)?

**Review by wjthieme — COMMENTED** [review]
Awesome! Great first start! Can you try to make the code snippets even simpler?
* Instead of functions write them as a script (newer versions of js support top-level awaits)
* Try to keep the snippet as simple and to the point as possible. i.e. for creating a wallet just keep it to creating the wallet, no need to store the keypair on file, read from env, etc.

**Comment by wjthieme** [comment]
Maybe instead of a separate section for `Legacy SDK` can we add it to the current guides as tabs (the same way we would also add rust high-level sdk in the future)?

File: docs/whirlpool/package.json
Comment (calintje) [reply docs/whirlpool/package.json:8]: It can create copies of components in your own repo, so you can add styling to them. Not really needed to have it as a script.

File: docs/whirlpool/docs/05-Legacy SDK/01-Developer Portal.md
Comment (calintje) [reply docs/whirlpool/docs/05-Legacy SDK/01-Developer Portal.md:1]: This is meant as a placeholder for the docs of the Legacy SDK. I will remove it from the sidebar for now.

File: docs/whirlpool/docs/03-Getting Started/05-Monitor/01-Splash Pool.md
Comment (calintje) [reply docs/whirlpool/docs/03-Getting Started/05-Monitor/01-Splash Pool.md:5]: The idea was to place all the 'fetch' functions (pools and positions) under the sidebar category "Monitor", but can also add them under "Create Pool" and "Provide Liquidity" respectively.

File: docs/whirlpool/docs/03-Getting Started/03-Provide Liquidity/01-Open Position/02-CLMM Pool.md
Comment (calintje) [reply docs/whirlpool/docs/03-Getting Started/03-Provide Liquidity/01-Open Position/02-CLMM Pool.md:2]: Agree

File: docs/whirlpool/docs/03-Getting Started/02-Create Pool/01-Splashpool.md
Comment (calintje) [reply docs/whirlpool/docs/03-Getting Started/02-Create Pool/01-Splashpool.md:57]: 🙏🏼

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (calintje) [reply docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:52]: Agree

File: docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md
Comment (calintje) [reply docs/whirlpool/docs/03-Getting Started/01-Environment Setup.md:60]: I will update this and all related comments.

**Review by calintje — COMMENTED** [review]
Thank you for the comments. Will push changes later today.

File: docs/whirlpool/docs/03-Getting Started/05-Monitor/01-Splash Pool.md
Comment (wjthieme) [reply docs/whirlpool/docs/03-Getting Started/05-Monitor/01-Splash Pool.md:5]: Yeah think it makes sense to have fetchPool functions together with create pool and fetchPosition function together with provide liquidity!

**Comment by calintje** [comment]
> Maybe instead of a separate section for `Legacy SDK` can we add it to the current guides as tabs (the same way we would also add rust high-level sdk in the future)?

I will add this when I port over the documentation from the developer portal.

File: docs/whirlpool/package.json
Comment (wjthieme) [reply docs/whirlpool/package.json]: If not really needed can we remove?

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:19]: Not 100% sure on this but no divergence loss on splash pools or just very little that it is negligible?

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:34]: Order of the mints matters. Maybe good to explain that here?

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:80]: Nice! 🙌🏼

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:86]: Since the functions will be different on different SDKs (rust, legacy, ts) would it make sense to just say: "You can use Splash Pools to set a up the pool ..."

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:94]: When opening a position you don't have to worry about the ratio so probably good to not even mention that here. For CL pools your liquidity will need to be around the current price (otherwise it is still not tradable) but we can add that in the doc for opening a position

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md:7]: nit: "insight in"

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md:11]: nit: "insight"

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:42]: I would put the `rpc` parameter in all lowercase

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md:15]: Think this one is now called `fetchConcentratedLiquidityPool`

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md:31]: Maybe good here to explain the `initialized` paramter

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md:38]: Slippage is currently based on amounts and not on price. i.e. if my expected ratios are:
tokenAEst: 100
tokenBEst: 1000

my max amounts spent will be:
tokenAMax: 101
tokenBMax: 1010

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md:59]: Maybe add a note that positions only earn fees if the price falls within you selected price range?

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md:61]: You can also set the default funder in the SDK so that you don't have to keep passing it to each function (similar to slippage tolerance)

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/02-Fetch Positions.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/02-Fetch Positions.md:7]: Should we rename this function to something like `fetchPositionsForOwner`?

Comment (wjthieme) [reply docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/02-Fetch Positions.md:7]: Since @orca-so/whirlpools-client has `fetchPosition(address)` and `fetchAllPositions(addresses) functions

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/04-Harvest.md
Comment (wjthieme) [inline docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/04-Harvest.md:7]: And potentially reinvest your earned fees into the position?

**Review by wjthieme — COMMENTED** [review]
Awesome!

File: docs/whirlpool/docs/03-Whirlpools SDK/01-Environment Setup.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/01-Environment Setup.md:32]: Not strictly needed. I will skip Typescript project initialization.

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:19]: Divergence loss is only mentioned in relation to Concentrated Liquidity Pools.

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:86]: Good point, I will apply this to all docs.

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/02-Fetch Pools.md:31]: Add explainer at section 1. Overview of Pool Fetching

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md:38]: Interesting. Is this only the case for positions, or for swaps as well?

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/02-Fetch Positions.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/02-Fetch Positions.md:7]: I agree that the name `fetchPositionsForOwner()` more accurately describes the functionality of this function.

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md:61]: In addition, I could also add this to the 'Environment Setup' doc.

On another note, I noticed that in the various function signatures the parameter that holds `FUNDER` is named `funder`, `authority`, or `signer`.

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (wjthieme) [reply docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:19]: Yeah but is there really no divergence loss on splash pools (or full range positions in CL pools)?

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (wjthieme) [reply docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md:38]: Yep also swaps

File: docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md
Comment (wjthieme) [reply docs/whirlpool/docs/03-Whirlpools SDK/03-Position Management/01-Open Position.md]: Yeah they are used interchangeably a little bit but generally they have a slightly different meaning:
* Authority: a wallet that has special privileges (i.e. the owner of a position, the owner of a token account, etc.)
* Funder: the wallet that deposits liquidity
* Payer: short for fee-payer (the one that pays for the transaction fee)

Let me know if you think we should change the param names

File: docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md
Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:19]: Ahh, got it. I think I misread your first comment and interpreted it as a statement - that there's no divergence loss for splash pools. But indeed, you could have divergence loss in full range positions. I will update!

Comment (calintje) [reply docs/whirlpool/docs/03-Whirlpools SDK/02-Whirlpool Management/01-Create Pool.md:19]: I removed that bit from this doc and updated the explainer on divergence loss in the doc Open Position.md, where I think it's more appropriate.

**Review by wjthieme — APPROVED** [review]
Approved but let's hold off on merging until #201 is merged and published too 🚀

**Comment by calintje** [comment]
> Approved but let's hold off on merging until #201 is merged and published too 🚀

Awesome! Thank you for reviewing! 🙏🏼

**Review by MrG9090 — APPROVED** [review]

**Pull request merged**
